test/integration_test/execute_rule/nonregistered_rule.js

```diff
@@ -0,0 +1,168 @@
+ *        - Organization contract is deployed and worker is set.
+ *        - TokenRules contract is deployed.
+ *          TransferRule contract is deployed and it is registered in TokenRules.
+ *        - TokenHolder contract is deployed by providing the wallets and
+ *           required confirmations.
+ *        - Using EIP20TokenMock's setBalance method,tokens are provided to TH.
+ *        - A contract i.e. TransferRule is deployed and is not registered in
+ *          TokenRules
+ *        - We generate executable data for TransferRule contract's(which was not
+ *          registered in TokenRules)transferFrom method.
+ *        - Relayer calls executeRule method of tokenholder contract.
+ *           After it's execution below verifications are done:
+ *            - RuleExecuted event.
+ *            - tokenholder balance.
+ *            - 'to' address balance.
+ */
+const EthUtils = require('ethereumjs-util'),
+  utils = require('../../test_lib/utils'),
+  AccountsProvider = utils.AccountProvider,
+  ExecuteRuleUtils = require('./utils'),
+  BN = require('bn.js'),
+  { Event } = require('../../test_lib/event_decoder');
+
+contract('TokenHolder::executeRule', async (accounts) => {
+
+  let accountProvider,
+    tokenHolder,
+    wallet1,
+    eip20TokenMock,
+    transferRule,
+    tokenRules,
+    ephemeralPrivateKey1,
+    ephemeralKeyAddress1,
+    keyData,
+    totalBalance = 500,
+    transferRule2;
+
+  describe('ExecuteRule integration test', async () => {
+
+    it('Should fail when non-registered rule calls executerule', async () => {
+
+      accountProvider = new AccountsProvider(accounts);
+
+      ( {
+        tokenHolder,
+        wallet1,
+        eip20TokenMock,
+        transferRule,
+        tokenRules,
+      } = await ExecuteRuleUtils.setup(accountProvider));
+
+      await eip20TokenMock.setBalance(tokenHolder.address, totalBalance);
+
+      ephemeralPrivateKey1 = '0xa8225c01ceeaf01d7bc7c1b1b929037bd4050967c5730c0b854263121b8399f3';
+      ephemeralKeyAddress1 = '0x62502C4DF73935D0D10054b0Fb8cC036534C6fb0';
+
+      let currentBlockNumber = await web3.eth.getBlockNumber(),
+        expirationHeight = currentBlockNumber + 50,
+        spendingLimit = 200;
+
+      await tokenHolder.submitAuthorizeSession(
+        ephemeralKeyAddress1,
+        spendingLimit,
+        expirationHeight,
+        { from: wallet1 },
+      );
+
+      keyData = await tokenHolder.ephemeralKeys(
+        ephemeralKeyAddress1,
+      );
+
+      let currentNonce = keyData.nonce,
+        amountTransferred = 50;
+
+      let nextAvailableNonce = currentNonce.toNumber() + 1;
+      const to = accountProvider.get();
+
+      const transferFromExecutable = await ExecuteRuleUtils.generateTransferFromExecutable(
+        tokenHolder.address,
+        to,
+        new BN(amountTransferred),
+      );
+
+      // It is not registered in TokenRules.
+      transferRule2 = await ExecuteRuleUtils.transferRule(tokenRules.address);
```

Comment:
we can rename transferRule2 to "nonRegisteredRule" for better clarity.

Reply:
Done.